MozillaFoundation/foundation.mozilla.org
# Fix blog newsletter signup id on translated pages

# Description

The blog newsletter signup is passing invalid id's on translated pages. Using `context` to get an "unlocalized" signup_id instead of the `| unlocalize` filter will address the bug.

# To Test
1) Grab a copy of staging/production data via `inv copy-prod-db`
2) View an `en` version of a blog post with a newsletter i.e. `/en/blog/how-to-tell-chat-gpt-generated-text/`
3) Inspect the newsletter and note the `data-signup-id` and note the id is 1
4) View the `es` translation of that same blog post i.e. `/es/blog/how-to-tell-chat-gpt-generated-text/`
5) Inspect the newsletter and note the `data-signup-id` is now 3.
6) Check out this branch locally and repeat steps 2-5, but note that the `data-signup-id` remains 1.


┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-1281)
